src/components/MenuPanel/MenuPanel.js

```diff
@@ -65,11 +65,42 @@ function MenuPanel({
   const [systemAppsOpened, setSystemAppsOpened] = useState(
     systemAppsOpenedState.isOpen()
   )
-  const [systemAppsToggled, setSystemAppsToggled] = useState(false)
+  const [showSystemsAppsAnimation, setShowSystemsAppsAnimation] = useState(
+    false
+  )
+
+  const appGroups = useMemo(
+    () =>
+      appInstanceGroups
+        .filter(appGroup => appGroup.hasWebApp)
+        .map(appGroup => ({
+          ...appGroup,
+          icon: <AppIcon app={appGroup.app} />,
+        })),
+    [appInstanceGroups]
+  )
+
+  const showConsole = consoleVisible || activeInstanceId === 'console'
+  const menuApps = [APP_HOME, appGroups]
+  const systemApps = [
+    APP_PERMISSIONS,
+    APP_APPS_CENTER,
+    APP_ORGANIZATION,
+    ...(showConsole ? [APP_CONSOLE] : []),
+  ]
+
+  const isSystemAppActive = useMemo(
```

Comment:
Moving this to memoized state (rather of computing it) simplifies the corresponding `useEffect()`.